diff --git src/common.h src/common.h
index c91d349..461b2a5 100644
--- src/common.h
+++ src/common.h
@@ -24,7 +24,7 @@
 #define LDCONFIG_PATH             "/sbin/ldconfig"
 #define LDCONFIG_ALT_PATH         "/sbin/ldconfig.real"
 
-#define LIB_DIR                   "/lib64"
+#define LIB_DIR                   "/usr/local/glibc/lib"
 #define USR_BIN_DIR               "/usr/bin"
 #define USR_LIB_DIR               "/usr/lib64"
 #define USR_LIB32_DIR             "/usr/lib32"
@@ -33,7 +33,7 @@
 #if defined(__x86_64__)
 # define LIB_ARCH                 LD_X8664_LIB64
 # define LIB32_ARCH               LD_I386_LIB32
-# define USR_LIB_MULTIARCH_DIR    "/usr/lib/x86_64-linux-gnu"
+# define USR_LIB_MULTIARCH_DIR    "/usr/local/lib"
 # define USR_LIB32_MULTIARCH_DIR  "/usr/lib/i386-linux-gnu"
 # if !defined(__NR_execveat)
 #  define __NR_execveat 322
