--- a/clients/upsmon.c
+++ b/clients/upsmon.c
@@ -29,6 +29,7 @@
 # include <sys/socket.h>
 # include <unistd.h>
 # include <fcntl.h>
+# include <spawn.h>
 #else	/* WIN32 */
 # include "wincompat.h"
 #endif	/* WIN32 */
@@ -190,7 +191,8 @@
 /* pre-declare internal methods */
 static int get_var(utype_t *ups, const char *var, char *buf, size_t bufsize);
 static void set_alarm(void);
 static void clear_alarm(void);
+static int runcmd(const char *cmd);

 static void setflag(int *val, int flag)
 {
@@ -1049,7 +1051,7 @@
 
 		upsdebugx(1, "%s: upsmon mono-process: Calling shutdown command: %s",
 			__func__, shutdowncmd);
-		sret = system(shutdowncmd);
+		sret = runcmd(shutdowncmd);
 
 		if (sret != 0)
 			upslogx(LOG_ERR, "Unable to call shutdown command: %s",
@@ -3491,6 +3493,22 @@
 	exit(EXIT_SUCCESS);
 }
 
+static int runcmd(const char *cmd)
+{
+	// TODO: Support tokenizing `cmd`.
+	const char *argv[] = {cmd, NULL};
+	const char *env[] = {NULL};
+
+	pid_t pid = 0;
+	int sret = posix_spawn(&pid, cmd, NULL, NULL, (char * const *)(argv), (char * const *)(env));
+	if (sret != 0)
+		return -1;
+
+	int status;
+	waitpid(pid, &status, 0);
+	return WEXITSTATUS(status);
+}
+
 #ifndef WIN32
 static void runparent(int fd)
 	__attribute__((noreturn));
@@ -3544,7 +3562,7 @@
 
 	upsdebugx(1, "%s: upsmon parent: Calling shutdown command: %s",
 		__func__, shutdowncmd);
-	sret = system(shutdowncmd);
+	sret = runcmd(shutdowncmd);
 
 	if (sret != 0)
 		upslogx(LOG_ERR, "upsmon parent: Unable to call shutdown command: %s",
