diff --git cmd/nvidia-container-runtime-hook/hook_config.go cmd/nvidia-container-runtime-hook/hook_config.go
index 9913ccc..405aa92 100644
--- cmd/nvidia-container-runtime-hook/hook_config.go
+++ cmd/nvidia-container-runtime-hook/hook_config.go
@@ -11,7 +11,7 @@ import (
 )
 
 const (
-	configPath = "/etc/nvidia-container-runtime/config.toml"
+	configPath = "/usr/local/etc/nvidia-container-runtime/config.toml"
 	driverPath = "/run/nvidia/driver"
 )
 

